src/screens/Wallet.js

```diff
@@ -141,8 +142,10 @@ class Wallet extends React.Component {
    * When user confirms the unregister of the token, hide the modal and execute it
    */
   unregisterConfirmed = () => {
-    const promise = tokens.unregisterToken(this.props.selectedToken);
+    const tokenUid = this.props.selectedToken;
+    const promise = tokens.unregisterToken(tokenUid);
```

Comment:
I know `promise` already existed, but why do we have this variable?

Why don't we just do something like:

```typescript
  tokens.unregisterToken(tokenUid) 
    .then(...)
```

Reply:
Good point. Since the function can only return a `promise`, whether for success or failure, I changed it to an `async` function to help static code analyzers identify this as well.
Refactored on f925b5c7e7a5df5ac8bc7d3c82b4ec6df9b9e0f5